pkg/limayaml/validate.go

```diff
@@ -108,6 +108,10 @@ func Validate(y LimaYAML, warn bool) error {
 		return errors.New("field `cpus` must be set")
 	}
 
+	if *y.CPUs > runtime.NumCPU() {
+		return fmt.Errorf("field `cpus` is set to %d, which is greater than the number of CPUs available (%d)", *y.CPUs, runtime.NumCPU())
+	}
```

Comment:
Wondering if the VM driver should automatically decrease the CPUs (with a warning)?

Reply:
We could (i have no strong opinion either way), but why do it in the driver and not in `FillDefault`? We set the default memory in there based on actual memory available, so why not implement the cap on the number of CPUs there as well?

Reply:
I just remembered: we may be executing `FillDefault` several times (we read the configurations of all instances during startup to look for things like shared network daemons, or shared additional disks). So any warnings generated while processing `lima.yaml` will be displayed multiple times, and we may be displaying warnings for other instances too.

So doing it in the driver might be preferable after all.

Reply:
Sorry, I have a question.
Am I correct in thinking this should be implemented in `vz_driver_darwin.go` and `qemu_driver.go`?
Thanks

Reply:
i'm able to generate a warning, but I'm not sure how to edit the yaml config to permanently set the number of cpu's.
setting `y.cpus` doesn't edit the yaml when viewed using `limactl edit`